dexter/src/main/java/com/karumi/dexter/DexterInstance.java

```diff
@@ -194,6 +196,9 @@ private PermissionStates getPermissionStates(Collection<String> pendingPermissio
    * https://github.com/Karumi/Dexter/issues/86
    */
   private int checkSelfPermission(Activity activity, String permission) {
+    if (forcePermissionRequest) {
+      return PackageManager.PERMISSION_DENIED;
```

Comment:
According to the logic flow, when `PERMISSION_DENIED` is returned, then the permission is explicitly requested by `AndroidPermissionService` class.